superset/assets/src/query/queryContextBuilder.spec.ts

```diff
@@ -0,0 +1,24 @@
+import * as datasourceBuilder from './datasourceBuilder';
+import build from './queryContextBuilder';
+import * as queryObjectBuilder from './queryObjectBuilder';
+
+describe('queryContextBuilder', () => {
+  const formData = {
+    datasource: '5__table',
+    groupby: ['foo'],
+  };
+  const datasourceBuilderSpy = jest.spyOn(datasourceBuilder, 'default');
+  const queryObjectBuilderSpy = jest.spyOn(queryObjectBuilder, 'default');
+
+  beforeEach(() => {
+    build(formData);
+  });
+
+  it('should call datasourceBuilder to build datasource', () => {
+    expect(datasourceBuilderSpy).toHaveBeenCalled();
```

Comment:
nit -- possibly more robust if we specify the # times called with `toHaveBeenCalledTimes(1)` but not huge deal. I think the spies would have to be reset per-call tho.

Reply:
I have question about where to put test files. I like having them side-by-side but both `incubator-superset` and `superset-ui` put test files in separate folders. Perhaps this PR can follow the same structure to put tests under `test`?

Reply:
Would we move everything under visualization to `superset-ui` eventually? If so, perhaps, we could put the query builder tests and the `buildQuery` tests for each visualization together in a dedicated directory (call it `superset-ui`?) under `spec/javascripts`? This way we could move them all together. Thoughts?

Reply:
sgtm. That's how I did it for the color modules.